docs/proposals/apiserver_watch.md

```diff
@@ -0,0 +1,132 @@
+## Abstract
+
+In the current system, all watch requests send to apiserver are in general
+redirected to etcd. This means that for every watch request to apiserver,
+apiserver opens a watch on etcd.
+
+The purpose of the proposal is to improve the overall performance of the system
+by solving the following problems:
+
+- having too many open watches on etcd
+- avoiding deserializing/converting the same objects multiple times in different
+watch results
```

Comment:
So I debate at times why we don't cache the entire contents of etc in the api-server, even O(10^3+) nodes, it's not that much data, vs. acting like an etcd pass through.